ingestion_program/ingestion.py

```diff
@@ -51,11 +47,14 @@ def install_from_whitelist(req_file):
             print(f"requested package {package} has invalid format, will install latest version (of {package_version[0]}) if allowed")
             package = package_version[0]
         elif len(package_version) == 2:
-            version = package_version[1]
-            if not VERSION_PATTERN.match(version):
-                # invalid format of version, don't use
-                print(f"requested package {package} has invalid version, will install latest version (of {package_version[0]}) if allowed")
-                package = package_version[0]
+            version_str = package_version[1]
+            Version(version_str)
+            # try:
+            #     Version(version_str)
+            # except InvalidVersion:
+            #     print(f"requested package {package} has invalid version, will install latest version (of {package_version[0]}) if allowed")
+            #     package = package_version[0]

```

Comment:
```suggestion
            version_str = package_version[1]
            try:
                 Version(version_str)
            except InvalidVersion:
                 exit(f"requested package {package} has invalid version, please check that {version_str} is the correct version of {package_version[0]}.")
            #     package = package_version[0]

```
This still needs to be wrapped in a `try` and the `InvalidVersion` needs to be caught. If we're not installing the default package version if in the whitelist, then the exception should have an exit on the invalid version (I think the suggestion should do it, please check).